src/init.cpp

```diff
@@ -1677,12 +1677,19 @@ bool AppInitMain(InitInterfaces& interfaces)
     }
 
     // ********************************************************* Step 11: import blocks
-
+    uint64_t bytes_needed = 0;
+    if (!chainActive.Height()) {
```

Comment:
I don't think this is the right check for first run. I think the best one is to check if data dir existed.

Reply:
Why ?
data dir could exist without any block downloaded and it would be necessary to check for available disk space too.

Reply:
Ok. But note that genesis is being loaded asynchronously.

Reply:
Maybe
```cpp
if (chainActive.Height() <= 1)
```
would be better then ?